odoo/addons/base/models/ir_http.py

```diff
@@ -192,7 +192,10 @@ def _pre_dispatch(cls, rule, args):
         # verify the default language set in the context is valid,
         # otherwise fallback on the company lang, english or the first
         # lang installed
-        env = request.env if request.env.uid else request.env['base'].with_user(SUPERUSER_ID).env
+        if request.env.uid:
+            env = request.env(context=request.env.context | {'lang': request.session.context.get('lang', request.default_lang())})
+        else:
+            env = request.env['base'].with_user(SUPERUSER_ID).env
         request.update_context(lang=get_lang(env)._get_cached('code'))
```

Comment:
`get_lang` need to call `env['res.lang'].get_installed` which uses ORM query to read/fetch `res.lang` if cache miss 😢

Reply:
I fixed `get_lang` as follow: `env['res.lang'].with_context(lang='en_US').get_installed()]`